code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -137,6 +137,13 @@ What are the archived variables for?
 	sleeping_agent -= removed.sleeping_agent
 	agent_b -= removed.agent_b
 
+	oxygen = max(0,oxygen)
+	nitrogen = max(0,nitrogen)
+	carbon_dioxide = max(0,carbon_dioxide)
+	toxins = max(0,toxins)
+	sleeping_agent = max(0,sleeping_agent)
+	agent_b = max(0,agent_b)
```

Comment:
instead of doing this, on the lines above do `gas_var = max(gas_var - removed.gas_var, 0)`